WatsonDeveloperCloud.xcodeproj/project.pbxproj

```diff
@@ -4106,6 +4119,191 @@
 		};
 /* End PBXResourcesBuildPhase section */
 
+/* Begin PBXShellScriptBuildPhase section */
+		CAEC7A671FE961C7002EF4FD /* Run SwiftLint */ = {
+			isa = PBXShellScriptBuildPhase;
+			buildActionMask = 2147483647;
+			files = (
+			);
+			inputPaths = (
+			);
+			name = "Run SwiftLint";
+			outputPaths = (
+			);
+			runOnlyForDeploymentPostprocessing = 0;
+			shellPath = /bin/sh;
+			shellScript = "if which swiftlint >/dev/null; then\n    swiftlint lint --config ${PROJECT_DIR}/.swiftlint.yml --no-cache --path Source/ConversationV1\n    swiftlint lint --config ${PROJECT_DIR}/Tests/.swiftlint.yml --no-cache --path Tests/ConversationV1Tests\nelse\n    echo \"SwiftLint does not exist, download from https://github.com/realm/SwiftLint\"\nfi";
```

Comment:
We need `--no-cache` because swiftlint does not understand the symlinking we do for RestKit.  I discovered this the hard way -- I spent several hours trying to understand why swiftlint was still reporting errors that I had fixed in RestKit, only to discover its caching mechanism was the cause.

The scheme may restrict the set of included source files, but I don't think it imposes a particular subdirectory for the build, so I think `--path` is required.

And sadly, when `--path` is specified, this is also the base path where swiftlint expects to find `.swiftlint.yml`.  This is another behavior of swiftlint that was not clearly documented and had to be determined empirically (see [this issue](https://github.com/realm/SwiftLint/pull/1745#issuecomment-320802347)).

Regarding the requirement for swiftlint, the script that runs it checks for it to be installed an simply prints a message when it is not present to recommend it be installed.

Reply:
When building the SDKs with Carthage, if SwiftLint is not installed, the message about installing SwiftLint will be printed _to the build log_ (along with all the other build messages), but not on the console.  So users would only see this message if the build fails and they start going through the build log to determine why it failed.

But it wouldn't hurt to make the message a little more clear, so I've changed it to:

>SwiftLint is not available. Install SwiftLint to obtain linter results.  See https://github.com/realm/SwiftLint